src/main/java/org/jboss/modules/xml/ModuleXmlParser.java

```diff
@@ -902,20 +911,31 @@ private static void parseNativeArtifact(final MavenResolver mavenResolver, final
             throw missingAttributes(reader, required);
         }
 
+        final SystemPropertyConditionBuilder conditionBuilder = new SystemPropertyConditionBuilder();
+        final Set<String> encountered = new HashSet<>();
         int eventType;
         for (;;) {
             eventType = reader.nextTag();
             switch (eventType) {
                 case END_TAG: {
                     try {
-                        createMavenNativeArtifactLoader(mavenResolver, name, reader, specBuilder);
+                        if (conditionBuilder.resolve()) {
```

Comment:
Just out of curiousity, why defer the resolution of the condition builder to the end tag, instead of just parsing it to a boolean right away?

Reply:
No real reason, I don't think it makes much difference either way.